soh/soh/Enhancements/randomizer/randomizer.cpp

```
@@ -4738,31 +4558,166 @@ void Randomizer::CreateCustomMessages() {
+                break;
+            case ITEM_MEDALLION_SPIRIT:
+                getItemID = RG_SPIRIT_MEDALLION;
+                break;
+            case ITEM_KOKIRI_EMERALD:
+                getItemID = RG_KOKIRI_EMERALD;
+                break;
+            case ITEM_GORON_RUBY:
+                getItemID = RG_GORON_RUBY;
+                break;
+            case ITEM_ZORA_SAPPHIRE:
+                getItemID = RG_ZORA_SAPPHIRE;
+                break;
+            case ITEM_SONG_LULLABY:
+                getItemID = RG_ZELDAS_LULLABY;
+                break;
+            case ITEM_SONG_SUN:
+                getItemID = RG_SUNS_SONG;
+                break;
+            case ITEM_SONG_EPONA:
+                getItemID = RG_EPONAS_SONG;
+                break;
+            case ITEM_SONG_STORMS:
+                getItemID = RG_SONG_OF_STORMS;
+                break;
+            case ITEM_SONG_TIME:
+                getItemID = RG_SONG_OF_TIME;
+                break;
+            case ITEM_SONG_SARIA:
+                getItemID = RG_SARIAS_SONG;
+                break;
+            case ITEM_SONG_MINUET:
+                getItemID = RG_MINUET_OF_FOREST;
+                break;
+            case ITEM_SONG_BOLERO:
+                getItemID = RG_BOLERO_OF_FIRE;
+                break;
+            case ITEM_SONG_SERENADE:
+                getItemID = RG_SERENADE_OF_WATER;
+                break;
+            case ITEM_SONG_NOCTURNE:
+                getItemID = RG_NOCTURNE_OF_SHADOW;
+                break;
+            case ITEM_SONG_REQUIEM:
+                getItemID = RG_REQUIEM_OF_SPIRIT;
+                break;
+            case ITEM_SONG_PRELUDE:
+                getItemID = RG_PRELUDE_OF_LIGHT;
+                break;
+            default:
+                // We should never get here. If this branch of code executes,
+                // then you've added an item to extendedVanillaGetItemTable that
+                // should be in randoGetItemTable.
```

maybe we should throw an exception here?

---

Probably, would need to decide how to handle that exception exactly. Am I just throwing it for a crash handler to take care of or will the coding calling this function need to handle it?

---

i think crash handler, based on the comment it's something that we should never expect to happen outside of a dev making a mistake, so i don't see a need to handle it anywhere else

---

Addressed in 4b7548e99867d47a750c3916268d20050cde6067